### DISCUSSION TITLE
[@alova/wormhole] 按requestBody.content 自动识别发送类型

### DISCUSSION THREAD

**Discussion by Lnncoco** [alovajs/alova#619] [Feature Request]
### What problem does the new feature need to solve?

swagger 配置中有注明是 application/x-www-form-urlencoded 类型的请求，目前发送的时候是 application/json ，需要手动组装传入。看 createApis 文件目前只对 Blob 类型做了处理，希望能按 swagger 已给出的信息自动处理。

### How important is it to you?

Can make it easier to develop

**Comment by JOU-amjs** [comment]
可能需要更具体的一些示例来补充下，我可能理解得更加深刻点？

**Reply by Lnncoco** [reply]
> 可能需要更具体的一些示例来补充下，我可能理解得更加深刻点？

swagger 上展示的发送类型是 content-type: application/x-www-form-urlencoded，但当前生成的请求包发出去的并不是预期的类型，导致后端报错。发送的请求头是 content-type: application/json，需要手动处理一下才能正常。
翻了下swagger json，在每个接口的 requestBody.content 字段下能够获取明确的定义，信息足够自动化处理。
明天找找例子，来补个图。

**Reply by Lnncoco** [reply]
![image](https://github.com/user-attachments/assets/1da502ba-bb83-4782-90dd-6dbab1c90679)
这是一个swagger标明请求是 application/x-www-form-urlencoded 类型的，在页面上点击 try 发出去的时候默认也会处理成对应类型
![image](https://github.com/user-attachments/assets/32d8dd1c-c34d-4606-b46e-332972e94056)
如果调用 Apis.login({  data: {  username: "xxx", password: "xxx" } }) 以对象的形式传参，内部不会根据swagger定义转换成 FormData 类型，会报错（因为发送的是 content-type: application/json），目前手动 new FormData 组装传入 data 才能正确发送，但感觉不太合理，明明可以省略这一步。

贴上去后发现图片好像无法放大，swagger里面的那个图是对应的json字段。

**Comment by MeetinaXD** [comment]
你使用的 requestAdapter 是什么？可以贴一下配置 alova 的代码。

**Reply by Lnncoco** [reply]
> 你使用的 requestAdapter 是什么？可以贴一下配置 alova 的代码。

fetchAdapter

```
export const alovaInstance = createAlova({
  id: 'apis',
  cacheFor: null,
  baseURL: import.meta.env.VITE_API_URL_PREFIX,
  statesHook: vueHook,
  requestAdapter: fetchAdapter(),
  beforeRequest: onAuthRequired(() => {}),
  responded: onResponseRefreshToken({
    // 请求成功返回处理逻辑
    onSuccess: async (response) => {
    },
    onError: async (err) => {
      console.error(`${err}`);
    },
  }),
});
```

**Comment by MeetinaXD** [comment]
了解，你可以使用 [XHR 适配器](https://alova.js.org/zh-CN/resource/request-adapter/xhr)。

**Comment by MeetinaXD** [comment]
@JOU-amjs 需要看看，fetch 适配器不支持`application/x-www-form-urlencoded` 算个bug吗？
